File: WebAPI/StoryBoard/DTO/UserProjectDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DTO;

public class UserProjectDto
{
    public int Id { get; set; }
    public int IdUser { get; set; }
    public int IdProject { get; set; }
    public string UserType { get; set; }
    public int AvailabilityTime { get; set; }
}

File: UI/StoryBoard/src/app/dtos/user-project-dto/user-project-dto.ts
export class UserProjectDto {
    id : number = 0;
    idUser : number = 0;
    idProject :number = 0;
    userType : string = "";
    availabilityTime : number = 0;

    static createFromFormValues(formValues: any): UserProjectDto {
        const userproject = new UserProjectDto();
        userproject.idUser = formValues.idUser;
        userproject.idProject = formValues.idProject;
        userproject.userType = formValues.userType;
        userproject.availabilityTime = formValues.availabilityTime;
        return userproject;
    }
}